errors.go

```diff
@@ -2,6 +2,17 @@ package sqlparser
 
 import "fmt"
 
+// ErrSyntaxError indicates a syntax error.
+type ErrSyntaxError struct {
```

Comment:
New error type. The parser sets through the [lexer](https://github.com/tablelandnetwork/go-sqlparser/pull/11/files#diff-95ead85843dcffb17425cf5129e674d63513c1465159ff5cc575d4aa03e99948R121)'s `Error` method this error when a syntax error occurs. And [here](https://github.com/tablelandnetwork/go-sqlparser/pull/11/files#diff-83eb8e32639d01cf443d6d8bde24c1c8be78766090d8c5f8586c36250cfedca6R37), you can see how the Parser method checks if a syntax error occurred.